core/transformer/src/test/standalone/IModelTransformer.test.ts

```diff
@@ -2024,6 +2026,91 @@ describe("IModelTransformer", () => {
     noSystemSchemasTransformer.dispose();
   });
 
+  it("handles long schema names and references to them", async function () {
+    const longSchema1Name = `ThisSchemaIs${"Long".repeat(100)}`;
+    assert(Buffer.from(longSchema1Name).byteLength > 255);
+    const longSchema2Name = `${longSchema1Name}ButEndsDifferently`;
+
+    if (process.platform !== "win32") {
+      // windows has no bound on path segment (file name) length, (it does have a bound on total path length),
```

Comment:
I think you changed this to limit to 255 chars on all platforms (an improvement IMO) so no need to check here?

Reply:
well this just proves that you can't write >255 byte file names on a platform level, it's not exercising my logic to limit it for schemas during export, so on windows this won't fail (there is no path segment limit), and removing that condition will fail the test on windows.

This assert below is more for asserting evidence of the source issue, I would prefer to keep it.